scio-core/src/main/scala/com/spotify/scio/coders/JodaSerializer.scala

```diff
@@ -0,0 +1,68 @@
+package com.spotify.scio.coders
+
+import com.esotericsoftware.kryo.io.{Input, Output}
+import com.esotericsoftware.kryo.{Kryo, Serializer}
+import org.joda.time.{LocalDate, LocalDateTime}
+import org.joda.time.chrono.ISOChronology
+
+class JodaLocalDateTimeSerializer extends Serializer[LocalDateTime] {
+  setImmutable(true)
+
+  def write(kryo: Kryo, output: Output, ldt: LocalDateTime): Unit = {
+    output.writeInt(ldt.getYear)
+    output.writeInt(ldt.getMonthOfYear)
+    output.writeInt(ldt.getDayOfMonth)
+    output.writeLong(ldt.getMillisOfDay)
+
+    val chronology = ldt.getChronology
+    if (chronology != null && chronology != ISOChronology.getInstanceUTC) {
+      sys.error(s"Unsupported chronology: $chronology")
+    }
+  }
+
+  def read(kryo: Kryo, input: Input, tpe: Class[LocalDateTime]): LocalDateTime = {
+    val y = input.readInt()
+    val m = input.readInt()
+    val d = input.readInt()
+    val millis = input.readLong()
+
+    new LocalDateTime(
+      y,
+      m,
+      d,
+      (millis / 3600000L).toInt,
+      ((millis % 3600000L) / 60000).toInt,
+      ((millis % 60000L) / 1000).toInt,
+      (millis % 1000L).toInt,
+      null
+    )
+  }
+}
+
+class JodaLocalDateSerializer extends Serializer[LocalDate] {
```

Comment:
The same comments as above:
 * use optimized versions of Int
 * use byte for small ints
 * read optimized
 * be explicit about constructor of `LocalDate`